esp/src/src-react/comms/fileSpray.ts

```diff
@@ -0,0 +1,342 @@
+            default:
+                retVal += path[i];
+        }
+    }
+    return retVal;
+};
+
+export interface DropZoneMachine {
+    Netaddress: string;
+    ConfigNetaddress: string;
+    Directory: string;
+    OS: number;
+}
+
+export interface DropZone {
+    Name: string;
+    Path: string;
+    Linux: string;
+    machine: DropZoneMachine;
+    TpMachines?: {
+        TpMachine: DropZoneMachine[];
+    };
+}
+
+export interface FileListItem {
+    calculatedID: string;
+    name: string;
+    isDir: boolean;
+    filesize?: number;
+    modifiedtime?: string;
+    NetAddress: string;
+    OS: number;
+    fullPath: string;
+    fullFolderPath: string;
+    DropZone: DropZone;
+    displayName: string;
+    type: string;
+    _isUserFile?: boolean;
+    getLogicalFile(): string;
+}
+
+export type LandingZonesStore = BaseStore<any, DropZone>;
+
+export function CreateLandingZonesStore(): LandingZonesStore {
+    const store = new Paged<any, DropZone>({
+        start: "PageStartFrom",
+        count: "PageSize"
+    }, "calculatedID", (request, abortSignal): Thenable<{ data: DropZone[], total: number }> => {
+        return topologyService.TpDropZoneQuery({
+            ...request,
+            ECLWatchVisibleOnly: true
+        }).then(response => {
+            const dropZones = response?.TpDropZones?.TpDropZone || [];
+            const data = dropZones.map((zone: any): DropZone => {
+                const processedZone = {
+                    ...zone,
+                    calculatedID: zone.Path + "_" + zone.Name,
+                    displayName: zone.Name,
+                    type: "dropzone",
+                    OS: zone.Linux === "true" ? 2 : 0,
+                    fullPath: zone.Path + (zone.Path && !zone.Path.endsWith("/") ? "/" : ""),
+                    DropZone: zone
+                };
+                return processedZone;
+            });
+
+            return {
+                data,
+                total: data.length
+            };
+        }).catch(err => {
+            const message = err?.Message ?? err?.message ?? err?.Exception?.[0]?.Message;
+            if (message === nlsHPCC.GridAbortMessage) {
+                logger.debug(message);
+            } else {
+                logger.error(err);
+            }
+            return {
+                data: [],
+                total: 0
+            };
+        });
+    });
+
+    const userAddedFiles: { [key: string]: FileListItem } = {};
+
+    (store as any).addUserFile = (file: FileListItem) => {
+        file._isUserFile = true;
+        userAddedFiles[file.calculatedID] = file;
+    };
+
+    (store as any).removeUserFile = (file: FileListItem) => {
+        delete userAddedFiles[file.calculatedID];
+    };
+
+    return store;
+}
+
+export type LandingZonesFilterStore = BaseStore<any, FileListItem>;
+
+export function CreateLandingZonesFilterStore(dropZone: DropZone): LandingZonesFilterStore {
+    const store = new Paged<any, FileListItem>({
```

Comment:
The any here could be pulled from comms (DropZoneFileSearch)?

Reply:
I think where I went wrong was using the Paged store..? None of these APIs return pages of data, I don't think. But it was the store that had the data fetch functions already. Need to think about this. Can probably just be Memory and add my own data loading?